In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## 투수

In [2]:
pitcher_data_2016 = pd.read_csv('C:/Users/user/Desktop/ESC_20_2/ESC_Summer_Team5/data/cleansing_data/team_pitcher/team_pitcher_2016.csv')
pitcher_data_2016.columns

Index(['G_ID', 'GDAY_DS', 'T_ID', 'VS_T_ID', 'HEADER_NO', 'TB_SC', 'CG_CK',
       'WLS', 'HOLD', 'INN2', 'BF', 'PA', 'AB', 'HIT', 'H2', 'H3', 'HR', 'SB',
       'CS', 'SH', 'SF', 'BB', 'IB', 'HP', 'KK', 'GD', 'WP', 'BK', 'ERR', 'R',
       'ER', 'P_WHIP_RT', 'P2_WHIP_RT', 'CB_WHIP_RT', 'H1', 'CUM_ER',
       'CUM_INN2', 'CUM_KK', 'CUM_BB', 'CUM_HIT', 'CUM_HR', 'CUM_AB', 'CUM_SF',
       'CUM_HP', 'CUM_R', 'CUM_H1', 'CUM_H2', 'CUM_H3', 'cFIP', 'K/9', 'BB/9',
       'K/BB', 'OAVG', 'OOBP', 'OSLG', 'OOPS', 'WHIP', 'BABIP', 'DICE', 'ERA',
       'RA9', 'FIP', 'kFIP', 'HR/9', 'H/9'],
      dtype='object')

In [3]:
pitcher_data_2016_x = pitcher_data_2016.loc[:,['R','ER','P_WHIP_RT','K/9','BB/9','K/BB','OAVG','OOBP','OSLG','OOPS','WHIP','BABIP','DICE',
                                               'ERA','RA9','FIP','kFIP','HR/9','H/9']]
pitcher_data_2016_x.corr()

R        ER  P_WHIP_RT       K/9      BB/9      K/BB  \
R          1.000000  0.959281   0.607972 -0.053454  0.128143 -0.145794   
ER         0.959281  1.000000   0.605315 -0.042834  0.116062 -0.130488   
P_WHIP_RT  0.607972  0.605315   1.000000  0.018241  0.122659 -0.100473   
K/9       -0.053454 -0.042834   0.018241  1.000000  0.133747  0.472990   
BB/9       0.128143  0.116062   0.122659  0.133747  1.000000 -0.737978   
K/BB      -0.145794 -0.130488  -0.100473  0.472990 -0.737978  1.000000   
OAVG       0.190955  0.195092   0.111158 -0.462781  0.205519 -0.452007   
OOBP       0.214403  0.208706   0.152602 -0.207264  0.743701 -0.731715   
OSLG       0.206562  0.210533   0.111000 -0.375850  0.238486 -0.483034   
OOPS       0.225973  0.226594   0.135610 -0.341825  0.449061 -0.615782   
WHIP       0.212737  0.207877   0.152720 -0.230977  0.729844 -0.746410   
BABIP      0.147469  0.157099   0.119835  0.018720  0.196355 -0.139204   
DICE       0.179191  0.171606   0.093937 -0.354969  0.635137 -0.772064   
ERA        0.226245  0.229413   0.153408 -0.188180  0.574959 -0.618611   
RA9        0.235185  0.227397   0.161059 -0.214298  0.637457 -0.676894   
FIP        0.180579  0.171473   0.093773 -0.323337  0.646370 -0.762606   
kFIP       0.178522  0.171324   0.099586 -0.163175  0.681770 -0.701771   
HR/9       0.151394  0.153111   0.065404 -0.132108  0.338205 -0.425269   
H/9        0.195128  0.198576   0.113594 -0.447519  0.185486 -0.435584   

               OAVG      OOBP      OSLG      OOPS      WHIP     BABIP  \
R          0.190955  0.214403  0.206562  0.225973  0.212737  0.147469   
ER         0.195092  0.208706  0.210533  0.226594  0.207877  0.157099   
P_WHIP_RT  0.111158  0.152602  0.111000  0.135610  0.152720  0.119835   
K/9       -0.462781 -0.207264 -0.375850 -0.341825 -0.230977  0.018720   
BB/9       0.205519  0.743701  0.238486  0.449061  0.729844  0.196355   
K/BB      -0.452007 -0.731715 -0.483034 -0.615782 -0.746410 -0.139204   
OAVG       1.000000  0.787663  0.824821  0.876366  0.814714  0.826211   
OOBP       0.787663  1.000000  0.688460  0.861394  0.983109  0.683021   
OSLG       0.824821  0.688460  1.000000  0.961429  0.722530  0.523348   
OOPS       0.876366  0.861394  0.961429  1.000000  0.878849  0.625549   
WHIP       0.814714  0.983109  0.722530  0.878849  1.000000  0.690025   
BABIP      0.826211  0.683021  0.523348  0.625549  0.690025  1.000000   
DICE       0.490307  0.700961  0.744891  0.787508  0.715275  0.091342   
ERA        0.735382  0.867006  0.816230  0.900440  0.856734  0.542386   
RA9        0.733208  0.901618  0.789923  0.895142  0.892272  0.536141   
FIP        0.458535  0.700646  0.726245  0.774329  0.700982  0.069695   
kFIP       0.399039  0.685172  0.699838  0.749967  0.682871  0.073864   
HR/9       0.377786  0.446079  0.766568  0.706028  0.470038  0.029748   
H/9        0.993605  0.770880  0.832627  0.875468  0.807127  0.822288   

               DICE       ERA       RA9       FIP      kFIP      HR/9  \
R          0.179191  0.226245  0.235185  0.180579  0.178522  0.151394   
ER         0.171606  0.229413  0.227397  0.171473  0.171324  0.153111   
P_WHIP_RT  0.093937  0.153408  0.161059  0.093773  0.099586  0.065404   
K/9       -0.354969 -0.188180 -0.214298 -0.323337 -0.163175 -0.132108   
BB/9       0.635137  0.574959  0.637457  0.646370  0.681770  0.338205   
K/BB      -0.772064 -0.618611 -0.676894 -0.762606 -0.701771 -0.425269   
OAVG       0.490307  0.735382  0.733208  0.458535  0.399039  0.377786   
OOBP       0.700961  0.867006  0.901618  0.700646  0.685172  0.446079   
OSLG       0.744891  0.816230  0.789923  0.726245  0.699838  0.766568   
OOPS       0.787508  0.900440  0.895142  0.774329  0.749967  0.706028   
WHIP       0.715275  0.856734  0.892272  0.700982  0.682871  0.470038   
BABIP      0.091342  0.542386  0.536141  0.069695  0.073864  0.029748   
DICE       1.000000  0.775909  0.790599  0.993285  0.974211  0.875727   
ERA        0.775909  1.000000  0.981562  0.787637  0.7

In [4]:
pitcher_data_2017 = pd.read_csv('C:/Users/user/Desktop/ESC_20_2/ESC_Summer_Team5/data/cleansing_data/team_pitcher/team_pitcher_2017.csv')
pitcher_data_2017_x = pitcher_data_2017.loc[:,['R','ER','P_WHIP_RT','K/9','BB/9','K/BB','OAVG','OOBP','OSLG','OOPS','WHIP','BABIP','DICE',
                                               'ERA','RA9','FIP','kFIP','HR/9','H/9']]

pitcher_data_2018 = pd.read_csv('C:/Users/user/Desktop/ESC_20_2/ESC_Summer_Team5/data/cleansing_data/team_pitcher/team_pitcher_2018.csv')
pitcher_data_2018_x = pitcher_data_2018.loc[:,['R','ER','P_WHIP_RT','K/9','BB/9','K/BB','OAVG','OOBP','OSLG','OOPS','WHIP','BABIP','DICE',
                                               'ERA','RA9','FIP','kFIP','HR/9','H/9']]

pitcher_data_2019 = pd.read_csv('C:/Users/user/Desktop/ESC_20_2/ESC_Summer_Team5/data/cleansing_data/team_pitcher/team_pitcher_2019.csv')
pitcher_data_2019_x = pitcher_data_2019.loc[:,['R','ER','P_WHIP_RT','K/9','BB/9','K/BB','OAVG','OOBP','OSLG','OOPS','WHIP','BABIP','DICE',
                                               'ERA','RA9','FIP','kFIP','HR/9','H/9']]

pitcher_data_2020 = pd.read_csv('C:/Users/user/Desktop/ESC_20_2/ESC_Summer_Team5/data/cleansing_data/team_pitcher/team_pitcher_2020.csv')
pitcher_data_2020_x = pitcher_data_2020.loc[:,['R','ER','P_WHIP_RT','K/9','BB/9','K/BB','OAVG','OOBP','OSLG','OOPS','WHIP','BABIP','DICE',
                                               'ERA','RA9','FIP','kFIP','HR/9','H/9']]

pitcher_data_final_x = pd.concat([pitcher_data_2016_x, pitcher_data_2017_x],axis=0)
pitcher_data_final_x = pd.concat([pitcher_data_final_x, pitcher_data_2018_x],axis=0)
pitcher_data_final_x = pd.concat([pitcher_data_final_x, pitcher_data_2019_x],axis=0)
pitcher_data_final_x = pd.concat([pitcher_data_final_x, pitcher_data_2020_x],axis=0)

pitcher_data_final_x.tail()

R  ER  P_WHIP_RT       K/9      BB/9      K/BB      OAVG      OOBP  \
635  1   1   0.600000  6.810665  3.219293  2.115578  0.270220  0.334591   
636  8   8   2.142857  7.614960  3.128755  2.433862  0.266827  0.331747   
637  4   4   3.000000  6.671145  3.359229  1.985915  0.288546  0.352546   
638  3   3   1.500000  7.053770  4.292227  1.643382  0.270626  0.356889   
639  4   4   1.800000  6.572165  2.814433  2.335165  0.279930  0.339592   

         OSLG      OOPS      WHIP     BABIP      DICE       ERA       RA9  \
635  0.421692  0.756284  1.396645  0.302632  4.242061  4.756141  5.192930   
636  0.399038  0.730786  1.368486  0.318467  3.509503  4.353771  4.800736   
637  0.419383  0.771930  1.521028  0.332785  3.730724  4.983645  5.551402   
638  0.407072  0.763961  1.523670  0.312791  4.213326  4.844535  5.381064   
639  0.416373  0.755966  1.405498  0.316639  3.974227  4.762887  5.195876   

          FIP      kFIP      HR/9        H/9  
635  4.954303  5.840462  1.164769   9.350509  
636  4.224383  5.158779  0.794605   9.187615  
637  4.461150  5.286500  0.757009  10.330023  
638  4.991168  5.878369  0.931034   9.420807  
639  4.738175  5.583536  1.036082   9.835052

In [5]:
pitcher_data_final_x.corr()

R        ER  P_WHIP_RT       K/9      BB/9      K/BB  \
R          1.000000  0.964062   0.571641 -0.018537  0.097155 -0.083853   
ER         0.964062  1.000000   0.559796 -0.020861  0.087697 -0.078340   
P_WHIP_RT  0.571641  0.559796   1.000000 -0.006376  0.075403 -0.063176   
K/9       -0.018537 -0.020861  -0.006376  1.000000  0.061941  0.403789   
BB/9       0.097155  0.087697   0.075403  0.061941  1.000000 -0.715611   
K/BB      -0.083853 -0.078340  -0.063176  0.403789 -0.715611  1.000000   
OAVG       0.201422  0.203176   0.115927 -0.198314  0.136144 -0.194072   
OOBP       0.209780  0.206229   0.127785 -0.109975  0.612445 -0.517207   
OSLG       0.225153  0.226738   0.117191 -0.043143  0.183169 -0.147157   
OOPS       0.233157  0.232957   0.128165 -0.070222  0.351414 -0.291540   
WHIP       0.212238  0.207315   0.133234 -0.091043  0.642680 -0.522813   
BABIP      0.160332  0.161807   0.099700  0.116863  0.113722 -0.045445   
DICE       0.196235  0.193118   0.106194 -0.179526  0.557475 -0.470601   
ERA        0.246221  0.249091   0.149687 -0.072922  0.443369 -0.370247   
RA9        0.252409  0.243265   0.157142 -0.069444  0.497195 -0.404129   
FIP        0.203528  0.201433   0.106548 -0.112306  0.458917 -0.377736   
kFIP       0.201824  0.199569   0.105784  0.036571  0.456939 -0.308441   
HR/9       0.176134  0.177499   0.085937  0.141888  0.161241 -0.028724   
H/9        0.205592  0.205978   0.118750 -0.162325  0.117584 -0.165681   

               OAVG      OOBP      OSLG      OOPS      WHIP     BABIP  \
R          0.201422  0.209780  0.225153  0.233157  0.212238  0.160332   
ER         0.203176  0.206229  0.226738  0.232957  0.207315  0.161807   
P_WHIP_RT  0.115927  0.127785  0.117191  0.128165  0.133234  0.099700   
K/9       -0.198314 -0.109975 -0.043143 -0.070222 -0.091043  0.116863   
BB/9       0.136144  0.612445  0.183169  0.351414  0.642680  0.113722   
K/BB      -0.194072 -0.517207 -0.147157 -0.291540 -0.522813 -0.045445   
OAVG       1.000000  0.849898  0.844966  0.897929  0.840142  0.905361   
OOBP       0.849898  1.000000  0.754649  0.890160  0.978190  0.771700   
OSLG       0.844966  0.754649  1.000000  0.970722  0.760377  0.669226   
OOPS       0.897929  0.890160  0.970722  1.000000  0.886154  0.747259   
WHIP       0.840142  0.978190  0.760377  0.886154  1.000000  0.755716   
BABIP      0.905361  0.771700  0.669226  0.747259  0.755716  1.000000   
DICE       0.472680  0.652594  0.747721  0.758165  0.675745  0.180170   
ERA        0.789120  0.837142  0.866373  0.908126  0.862390  0.633634   
RA9        0.768613  0.849791  0.838485  0.893389  0.877116  0.617307   
FIP        0.514322  0.660234  0.783527  0.785827  0.651065  0.252111   
kFIP       0.488650  0.640119  0.787265  0.781059  0.635134  0.268851   
HR/9       0.432878  0.422772  0.801188  0.711159  0.446033  0.212756   
H/9        0.991585  0.829751  0.854550  0.897209  0.836389  0.898195   

               DICE       ERA       RA9       FIP      kFIP      HR/9  \
R          0.196235  0.246221  0.252409  0.203528  0.201824  0.176134   
ER         0.193118  0.249091  0.243265  0.201433  0.199569  0.177499   
P_WHIP_RT  0.106194  0.149687  0.157142  0.106548  0.105784  0.085937   
K/9       -0.179526 -0.072922 -0.069444 -0.112306  0.036571  0.141888   
BB/9       0.557475  0.443369  0.497195  0.458917  0.456939  0.161241   
K/BB      -0.470601 -0.370247 -0.404129 -0.377736 -0.308441 -0.028724   
OAVG       0.472680  0.789120  0.768613  0.514322  0.488650  0.432878   
OOBP       0.652594  0.837142  0.849791  0.660234  0.640119  0.422772   
OSLG       0.747721  0.866373  0.838485  0.783527  0.787265  0.801188   
OOPS       0.758165  0.908126  0.893389  0.785827  0.781059  0.711159   
WHIP       0.675745  0.862390  0.877116  0.651065  0.635134  0.446033   
BABIP      0.180170  0.633634  0.617307  0.252111  0.268851  0.212756   
DICE       1.000000  0.757729  0.761217  0.949619  0.930065  0.843344   
ERA        0.757729  1.000000  0.977424  0.746695  0.7

### H/9 & OAVG  상관관계가 매우 높음 -> 둘 다 얼마나 안타를 맞았는지에 관한 지표
### HR/9 & kFIP 상관관계 높음
### DICE & kFIP & FIP 상관관계 매우 높음 
* DICE를 개량한 것이 FIP임
* kFIP가 FIP를 한국 KBO리그에 맞게 변형한 것이기 때문에 kFIP만 사용해도 될듯

### RA9 & ERA & OOPS 상관관계 높음
### WHIP & OOBP 상관관계 높음 -> 둘 다 얼마나 출루를 허용했는지에 관한 지표
* WHIP가 팀성적과 높은 상관관계 있다고 함

### OOPS & OOBP 상관관계 높음
### 전체적인 투수 지표를 보고 생각한 것은 OAVG ~ OOPS , BABIP는 같은 계산식을 타자에서 사용하니 투수에서는 제외하면 어떨지 생각

In [6]:
from sklearn.preprocessing import MinMaxScaler
scaler = MinMaxScaler()

## 투수 scaling OVAG ~ OOPS, BABIP를 제외한 나머지 값들을 다 0~1 사이로 scaling 함
col = ['K/9','BB/9','K/BB','WHIP','DICE','ERA','RA9','FIP','kFIP','HR/9','H/9']
pitcher_scaler_2016 = scaler.fit_transform(pitcher_data_2016.loc[:,col])
pitcher_scaler_2016 = pd.DataFrame(pitcher_scaler_2016)
pitcher_scaler_2016.columns = col
pitcher_data_2016[col] = pitcher_scaler_2016

pitcher_scaler_2017 = scaler.fit_transform(pitcher_data_2017.loc[:,col])
pitcher_scaler_2017 = pd.DataFrame(pitcher_scaler_2017)
pitcher_scaler_2017.columns = col
pitcher_data_2017[col] = pitcher_scaler_2017

pitcher_scaler_2018 = scaler.fit_transform(pitcher_data_2018.loc[:,col])
pitcher_scaler_2018 = pd.DataFrame(pitcher_scaler_2018)
pitcher_scaler_2018.columns = col
pitcher_data_2018[col] = pitcher_scaler_2018

pitcher_scaler_2019 = scaler.fit_transform(pitcher_data_2019.loc[:,col])
pitcher_scaler_2019 = pd.DataFrame(pitcher_scaler_2019)
pitcher_scaler_2019.columns = col
pitcher_data_2019[col] = pitcher_scaler_2019

pitcher_scaler_2020 = scaler.fit_transform(pitcher_data_2020.loc[:,col])
pitcher_scaler_2020 = pd.DataFrame(pitcher_scaler_2020)
pitcher_scaler_2020.columns = col
pitcher_data_2020[col] = pitcher_scaler_2020

In [7]:
pitcher_data_2016.to_csv('C:/Users/user/Desktop/ESC_20_2/team_pitcher_scaling_2016.csv')
pitcher_data_2017.to_csv('C:/Users/user/Desktop/ESC_20_2/team_pitcher_scaling_2017.csv')
pitcher_data_2018.to_csv('C:/Users/user/Desktop/ESC_20_2/team_pitcher_scaling_2018.csv')
pitcher_data_2019.to_csv('C:/Users/user/Desktop/ESC_20_2/team_pitcher_scaling_2019.csv')
pitcher_data_2020.to_csv('C:/Users/user/Desktop/ESC_20_2/team_pitcher_scaling_2020.csv')

## 타자

In [8]:
hitter_data_2016=pd.read_csv('C:/Users/user/Desktop/ESC_20_2/ESC_Summer_Team5/data/cleansing_data/team_batter/team_batter_revised_2016.csv')
hitter_data_2016.columns

Index(['G_ID', 'GDAY_DS', 'T_ID', 'VS_T_ID', 'HEADER_NO', 'TB_SC', 'PA', 'AB',
       'RBI', 'RUN', 'HIT', 'H2', 'H3', 'HR', 'SB', 'CS', 'SH', 'SF', 'BB',
       'IB', 'HP', 'KK', 'GD', 'ERR', 'LOB', 'P_HRA_RT', 'P_AB_CN', 'P_HIT_CN',
       'H1', 'CUM_HIT', 'CUM_AB', 'CUM_H1', 'CUM_H2', 'CUM_H3', 'CUM_HR',
       'CUM_BB', 'CUM_HP', 'CUM_SF', 'CUM_CS', 'CUM_GD', 'CUM_SB', 'CUM_SH',
       'CUM_IB', 'CUM_KK', 'CUM_PA', 'wBB', 'wHBP', 'w1B', 'w2B', 'w3B', 'wHR',
       'AVG', 'SLG', 'IsoP', 'OBP', 'OPS', 'RC', 'XR', 'wOBA', 'BABIP'],
      dtype='object')

In [9]:
hitter_data_2016_x = hitter_data_2016.loc[:,['RUN','P_HRA_RT','AVG','SLG','IsoP','OBP','OPS','RC','XR','wOBA','BABIP']]
hitter_data_2016_x.corr()

RUN  P_HRA_RT       AVG       SLG      IsoP       OBP  \
RUN       1.000000  0.662263  0.183527  0.180069  0.137169  0.202146   
P_HRA_RT  0.662263  1.000000  0.118697  0.105019  0.072184  0.125888   
AVG       0.183527  0.118697  1.000000  0.793497  0.476264  0.856623   
SLG       0.180069  0.105019  0.793497  1.000000  0.913035  0.712689   
IsoP      0.137169  0.072184  0.476264  0.913035  1.000000  0.455603   
OBP       0.202146  0.125888  0.856623  0.712689  0.455603  1.000000   
OPS       0.200082  0.119430  0.869982  0.970683  0.819414  0.860405   
RC        0.890018  0.508142  0.198180  0.193270  0.146423  0.197174   
XR        0.888697  0.512863  0.204109  0.204817  0.159132  0.209030   
wOBA      0.202764  0.123888  0.880323  0.922470  0.742822  0.918740   
BABIP     0.127193  0.097860  0.798575  0.475333  0.151563  0.657221   

               OPS        RC        XR      wOBA     BABIP  
RUN       0.200082  0.890018  0.888697  0.202764  0.127193  
P_HRA_RT  0.119430  0.508142  0.512863  0.123888  0.097860  
AVG       0.869982  0.198180  0.204109  0.880323  0.798575  
SLG       0.970683  0.193270  0.204817  0.922470  0.475333  
IsoP      0.819414  0.146423  0.159132  0.742822  0.151563  
OBP       0.860405  0.197174  0.209030  0.918740  0.657221  
OPS       1.000000  0.207969  0.220420  0.984963  0.570517  
RC        0.207969  1.000000  0.985972  0.206902  0.151833  
XR        0.220420  0.985972  1.000000  0.218860  0.147316  
wOBA      0.984963  0.206902  0.218860  1.000000  0.608450  
BABIP     0.570517  0.151833  0.147316  0.608450  1.000000

In [10]:
hitter_data_2017=pd.read_csv('C:/Users/user/Desktop/ESC_20_2/ESC_Summer_Team5/data/cleansing_data/team_batter/team_batter_revised_2017.csv')
hitter_data_2017_x = hitter_data_2017.loc[:,['RUN','P_HRA_RT','AVG','SLG','IsoP','OBP','OPS','RC','XR','wOBA','BABIP']]

hitter_data_2018=pd.read_csv('C:/Users/user/Desktop/ESC_20_2/ESC_Summer_Team5/data/cleansing_data/team_batter/team_batter_revised_2018.csv')
hitter_data_2018_x = hitter_data_2018.loc[:,['RUN','P_HRA_RT','AVG','SLG','IsoP','OBP','OPS','RC','XR','wOBA','BABIP']]

hitter_data_2019=pd.read_csv('C:/Users/user/Desktop/ESC_20_2/ESC_Summer_Team5/data/cleansing_data/team_batter/team_batter_revised_2019.csv')
hitter_data_2019_x = hitter_data_2019.loc[:,['RUN','P_HRA_RT','AVG','SLG','IsoP','OBP','OPS','RC','XR','wOBA','BABIP']]

hitter_data_2020=pd.read_csv('C:/Users/user/Desktop/ESC_20_2/ESC_Summer_Team5/data/cleansing_data/team_batter/team_batter_revised_2020.csv')
hitter_data_2020_x = hitter_data_2020.loc[:,['RUN','P_HRA_RT','AVG','SLG','IsoP','OBP','OPS','RC','XR','wOBA','BABIP']]

hitter_data_final_x = pd.concat([hitter_data_2016_x ,hitter_data_2017_x], axis = 0)
hitter_data_final_x = pd.concat([hitter_data_final_x ,hitter_data_2018_x], axis = 0)
hitter_data_final_x = pd.concat([hitter_data_final_x ,hitter_data_2019_x], axis = 0)
hitter_data_final_x = pd.concat([hitter_data_final_x ,hitter_data_2020_x], axis = 0)
hitter_data_final_x.tail()

RUN  P_HRA_RT       AVG       SLG      IsoP       OBP       OPS  \
635    7  0.272727  0.272513  0.436587  0.164074  0.357873  0.794460   
636    1  0.200000  0.271336  0.433210  0.161874  0.355905  0.789115   
637    9  0.266667  0.272562  0.435278  0.162716  0.357991  0.793269   
638    4  0.166667  0.271788  0.435310  0.163522  0.357254  0.792564   
639    3  0.333333  0.270916  0.433378  0.162461  0.356401  0.789779   

            RC      XR      wOBA     BABIP  
635   7.327179   6.714  0.355605  0.313580  
636   1.835556   1.468  0.353525  0.312880  
637  11.898723  10.270  0.355445  0.314593  
638   4.611111   4.882  0.355077  0.313090  
639   3.124324   3.238  0.354071  0.311485

In [11]:
hitter_data_final_x.corr()

RUN  P_HRA_RT       AVG       SLG      IsoP       OBP  \
RUN       1.000000  0.622845  0.215306  0.220655  0.174977  0.220848   
P_HRA_RT  0.622845  1.000000  0.144642  0.126139  0.085188  0.132614   
AVG       0.215306  0.144642  1.000000  0.802883  0.487621  0.895140   
SLG       0.220655  0.126139  0.802883  1.000000  0.911963  0.743801   
IsoP      0.174977  0.085188  0.487621  0.911963  1.000000  0.473261   
OBP       0.220848  0.132614  0.895140  0.743801  0.473261  1.000000   
OPS       0.234282  0.136126  0.884067  0.973023  0.816581  0.877942   
RC        0.887214  0.464655  0.239580  0.245952  0.195319  0.239126   
XR        0.886764  0.471477  0.245200  0.252614  0.201207  0.249444   
wOBA      0.226283  0.128956  0.882684  0.889083  0.694602  0.938517   
BABIP     0.163430  0.121377  0.889579  0.572075  0.225591  0.779838   

               OPS        RC        XR      wOBA     BABIP  
RUN       0.234282  0.887214  0.886764  0.226283  0.163430  
P_HRA_RT  0.136126  0.464655  0.471477  0.128956  0.121377  
AVG       0.884067  0.239580  0.245200  0.882684  0.889579  
SLG       0.973023  0.245952  0.252614  0.889083  0.572075  
IsoP      0.816581  0.195319  0.201207  0.694602  0.225591  
OBP       0.877942  0.239126  0.249444  0.938517  0.779838  
OPS       1.000000  0.258711  0.267044  0.960784  0.678944  
RC        0.258711  1.000000  0.985283  0.247085  0.190064  
XR        0.267044  0.985283  1.000000  0.256320  0.190682  
wOBA      0.960784  0.247085  0.256320  1.000000  0.692995  
BABIP     0.678944  0.190064  0.190682  0.692995  1.000000

### RC & XR -> 둘 다 득점 생산력 의미 (상관관계 매우 높음) => 1개만 써도 무방할듯
### wOBA & OPS이 상관관계 높음(0.960) -> 제거여부 고민 필요
* OPS(출루율 + 장타율) -> 가장 인정받는 타자 지표라고 함
* wOBA(가중출루율) -> OPS와 비슷하나 분석결과와 평가방법이 다르다고 함

### wOBA & OBP 상관관계 높음 -> wOBA는 OBP의 분자에 들어가는 값에 다른 가중치를 적용하였기 때문에
* -> wOBA만 써도 될듯

### OPS& SLG 상관관계가 높음(0.973) -> OPS가 SLG 포함하기 때문에 -> OPS만 써도 될듯
* OPS가 실제 공격 기여 수준과 높은 상관관계를 갖고 있다고 함

### IsOP & SLG 상관관계가 높음 -> IsOP 계산식에 SLG이 포함되기 때문
### 년도별로 바라보니 AVG&OBP 상관관계 높음(0.9정도)
### RUN(득점)과 상관관계가 높았던 것들 -> 'RC', 'XR', 'P_HRA_RT(득점권 타율)'

In [12]:
from sklearn.preprocessing import MinMaxScaler
scaler = MinMaxScaler()

## 타자에서는 XR, RC를 제외하고는 다 0~1사이이기에 이 둘을 0~!사이로 scaling하는 선택을 함 (시즌별로)
hitter_scaler_2016 = scaler.fit_transform(hitter_data_2016.loc[:,['RC','XR']])
hitter_scaler_2016 = pd.DataFrame(hitter_scaler_2016)
hitter_scaler_2016.columns = ['RC','XR']
hitter_data_2016[['RC','XR']] = hitter_scaler_2016

hitter_scaler_2017 = scaler.fit_transform(hitter_data_2017.loc[:,['RC','XR']])
hitter_scaler_2017 = pd.DataFrame(hitter_scaler_2017)
hitter_scaler_2017.columns = ['RC','XR']
hitter_data_2017[['RC','XR']] = hitter_scaler_2017

hitter_scaler_2018 = scaler.fit_transform(hitter_data_2018.loc[:,['RC','XR']])
hitter_scaler_2018 = pd.DataFrame(hitter_scaler_2018)
hitter_scaler_2018.columns = ['RC','XR']
hitter_data_2018[['RC','XR']] = hitter_scaler_2018

hitter_scaler_2019 = scaler.fit_transform(hitter_data_2019.loc[:,['RC','XR']])
hitter_scaler_2019 = pd.DataFrame(hitter_scaler_2019)
hitter_scaler_2019.columns = ['RC','XR']
hitter_data_2019[['RC','XR']] = hitter_scaler_2019

hitter_scaler_2020 = scaler.fit_transform(hitter_data_2020.loc[:,['RC','XR']])
hitter_scaler_2020 = pd.DataFrame(hitter_scaler_2020)
hitter_scaler_2020.columns = ['RC','XR']
hitter_data_2020[['RC','XR']] = hitter_scaler_2020


In [13]:
hitter_data_2016.to_csv('C:/Users/user/Desktop/ESC_20_2/team_hitter_scaling_2016.csv')
hitter_data_2017.to_csv('C:/Users/user/Desktop/ESC_20_2/team_hitter_scaling_2017.csv')
hitter_data_2018.to_csv('C:/Users/user/Desktop/ESC_20_2/team_hitter_scaling_2018.csv')
hitter_data_2019.to_csv('C:/Users/user/Desktop/ESC_20_2/team_hitter_scaling_2019.csv')
hitter_data_2020.to_csv('C:/Users/user/Desktop/ESC_20_2/team_hitter_scaling_2020.csv')